libs/neo4j/tests/unit_tests/chains/test_graph_qa.py

```diff
@@ -94,8 +100,8 @@ def test_graph_cypher_qa_chain_prompt_selection_3() -> None:
         qa_llm_kwargs={"memory": readonlymemory},
         allow_dangerous_requests=True,
     )
-    assert chain.qa_chain.prompt == CYPHER_QA_PROMPT  # type: ignore[union-attr]
-    assert chain.cypher_generation_chain.prompt == CYPHER_GENERATION_PROMPT
+    assert chain.qa_chain.first == CYPHER_QA_PROMPT  # type: ignore[attr-defined]
+    assert chain.cypher_generation_chain.first == CYPHER_GENERATION_PROMPT  # type: ignore[attr-defined]
```

Comment:
Should we ignore attr-defined here?

Reply:
Yes, both chains have the more generic `Runnable` class as their type hint. This class doesn't necessarily have `.first` method defined. However, in these tests, both chains are `RunnableSequence`s, a class which inherits from `Runnable` and which both chains are by default. So you need the attr-defined here

Reply:
Hmm I prefer not have type ignore unless absolutely necessary. Here in this case can we check that .first exists on the chains otherwise the test fails?

Reply:
Yeah, you're right it's not great practice. Have removed them now and added `hasattr` assertions to the tests.